cal/static/jsx/query.jsx

```diff
@@ -36,24 +37,52 @@ var RReset = React.createClass({
     }
 })
 
-var RFiltering = React.createClass({
+var RFilterCheckbox = React.createClass({
+    render: function() {
+        var user = this.props.user;
+        return (
+            <div className="filterForm">
+                <input type="checkbox" name="user" id={user.id} onChange={this.handleChange} defaultChecked="true"/>
+                <label for={user.id}>{user.name}</label>
+                <br />
+            </div>
+        );
+    },
+
+    handleChange: function(e) {
+        // If user is checked, add events.
+        if (e.target.checked) {
+            this.props.add(e.target.id);
+        }
+        else {
+            this.props.remove(e.target.id);
+        }
```

Comment:
I don't think this is very `react`-esque. `this.props` is supposed to immutable. See [this](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md).